### PR TITLE
Add profile to run integrations tests with Redis store simplementation

### PR DESCRIPTION
**Pull request by sbernard31**
To run the the integration-tests with redis implementation :

```
mvn clean install -Predis
```
or 
```
mvn clean install -Predis -Dredis.uri="redis://127.0.0.1:6379"   
```

### CODE REVIEW

File: leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/BootstrapIntegrationTestHelper.java [hunk 1]
```diff
@@ -46,9 +46,10 @@
  */
 public class BootstrapIntegrationTestHelper extends IntegrationTestHelper {
 
-    // public static final String SECURE_ENDPOINT_IDENTIFIER = "secureclient";
-    public static final String SECURE_PSK_ID = "Client_identity";
-    public static final byte[] SECURE_PSK_KEY = Hex.decodeHex("73656372657450534b".toCharArray());
+    public static final String GOOD_PSK_ID = "Good_Client_identity";
+    public static final byte[] GOOD_PSK_KEY = Hex.decodeHex("73656372657450534b".toCharArray());
+    public static final String BAD_PSK_ID = "Bad_Client_identity";
```
Comment (msangoi) [inline leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/BootstrapIntegrationTestHelper.java]: Not used

File: leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/ObserveTest.java [hunk 2]
```diff
@@ -44,10 +43,11 @@
 
 public class ObserveTest {
 
-    private IntegrationTestHelper helper = new IntegrationTestHelper();
+    protected IntegrationTestHelper helper = new IntegrationTestHelper();;
```
Comment (msangoi) [inline leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/ObserveTest.java]: double ;

File: leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/SecureIntegrationTestHelper.java [hunk 3]
```diff
@@ -254,21 +259,27 @@ public void createServerWithX509Cert(Certificate[] trustedCertificates) {
...
+                    protected void saveToFile() {
+                        // do not save to file
+                    }
+                });
 
         server = builder.build();
     }
+
+    @Override
+    public void dispose() {
```
Comment (msangoi) [inline leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/SecureIntegrationTestHelper.java]: super.dispose() to clear the endpoint identifier?

File: leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/RegistrationTest.java [hunk 4]
```diff
@@ -39,70 +33,73 @@
...
         helper.createClient();
-
     }
 
     @After
-    public void stop() {
+    public void stop() throws InterruptedException {
+        helper.client.stop(true);
         helper.server.stop();
+        helper.deregisterClient();
```
Comment (msangoi) [inline leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/RegistrationTest.java]: already done in helper.dispose()

**Comment by sbernard31** [comment]
Changes done.
I also add a parameter for redis URI :

```
mvn clean install -Predis -Dredis.uri="redis://127.0.0.1:6379"   
```

**Pull request merged**